common/src/main/java/edu/snu/nemo/common/ir/vertex/MetricCollectionBarrierVertex.java

```diff
@@ -29,7 +29,9 @@
  */
 public final class MetricCollectionBarrierVertex<T> extends IRVertex {
   // Partition ID to Size data
-  private final Map<String, List<T>> metricData;
+  private T metricData;
+  private final List<String> blockIds;
```

Comment:
Is there a particular reason for decoupling the same data structure into two?

Reply:
`blockIds` is not included in `metricData`. Metric data here is the stats collected to use in DataSkewRuntimePass. Block ids are used to designate the edges subjected to optimization.